app/src/main/java/de/westnordost/streetcomplete/quests/diet_type/AddHalal.kt

```diff
@@ -0,0 +1,34 @@
+package de.westnordost.streetcomplete.quests.diet_type
+
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.data.meta.updateWithCheckDate
+import de.westnordost.streetcomplete.data.osm.edits.update_tags.StringMapChangesBuilder
+import de.westnordost.streetcomplete.data.osm.osmquests.OsmFilterQuestType
+
+class AddHalal : OsmFilterQuestType<DietAvailability>() {
+
+    override val elementFilter = """
+        nodes, ways with
+        (
+          amenity ~ restaurant|cafe|fast_food|ice_cream
+          or shop ~ butcher|supermarket|ice_cream
+        )
+        and name and (
+          !diet:halal
+          or diet:halal != only and diet:halal older today -4 years
+        )
+    """
+    override val commitMessage = "Add halal status"
```

Comment:
I believe Halal should be capitalised? If so you should do so in the commit message (kosher isn't).
```suggestion
    override val commitMessage = "Add Halal status"
```

Reply:
Actually maybe not, Wikipedia hasn't (I missed a full stop). Although I see you have in quest_dietType_explanation_halal, it might be nice to be consistent.

Reply:
`quest_dietType_explanation_halal` is a sentence, so it should start with an uppercase letter.

Reply:
The "with Halal certification agencies" is just part of a sentance @FloEdelmann . Anyway presumably the fact @SMUsamaShah has merged this means it should be.